ecole-pratique-des-hautes-etudes-sciences-historiques-et-philologiques.csl

```diff
@@ -0,0 +1,620 @@
+            <text variable="number-of-pages" prefix=", " suffix=" p"/>
+          </if>
+          <else>
+            <text variable="number-of-pages" prefix=", "/>
+          </else>
+        </choose>
+      </else-if>
+      <else-if type="report entry-dictionary entry-encyclopedia chapter" match="any">
+        <group delimiter=" " prefix=", " suffix=".">
+          <label variable="page" form="short"/>
+          <text variable="page"/>
+        </group>
+      </else-if>
+      <else-if type="graphic personal_communication interview webpage" match="any"/>
+      <else-if type="article-journal article-magazine" match="any">
+        <group delimiter=" " font-style="normal" prefix=", ">
+          <group delimiter=" " suffix=".">
+            <label variable="page" form="short"/>
+            <text variable="page" form="short"/>
+          </group>
+        </group>
+      </else-if>
+      <else-if type="article-newspaper speech paper-conference" match="any">
+        <group font-style="normal">
+          <group delimiter=" " prefix=", ">
+            <label variable="page" form="short"/>
+            <text variable="page"/>
+          </group>
+        </group>
+      </else-if>
+    </choose> 
+  </macro>
+  <macro name="ISBN">
+    <text variable="ISBN" prefix=". ISBN : " suffix="."/>
+  </macro>
+  <macro name="archive">
+    <group suffix="." prefix=". ">
+      <group delimiter=", ">
+        <text variable="archive"/>
+        <text variable="archive_location"/>
+        <text variable="call-number"/>
+      </group>
+    </group>
+  </macro>
+  <macro name="volume">
+    <choose>
+      <if is-numeric="volume">
+        <text term="volume" form="short" suffix=". "/>
+        <text variable="volume"/>
+        <text variable="number-of-volumes" prefix="/"/>
+      </if>
+      <else>
+        <text variable="volume"/>
+        <text variable="number-of-volumes" suffix=" vol."/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="number">
+    <choose>
+      <if is-numeric="number">
+        <text variable="number" prefix="n° "/>
+      </if>
+      <else>
+        <text variable="number"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="issue">
+    <choose>
+      <if is-numeric="issue">
+        <text term="issue" form="short" suffix=" "/>
+        <text variable="issue"/>
+      </if>
+      <else>
+        <text variable="issue"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="interviewer">
+    <group>
+      <names variable="interviewer">
+        <label form="verb" prefix=" " suffix=" "/>
+        <name form="long" and="text" delimiter=", ">
+          <name-part name="family" font-variant="small-caps"/>
+        </name>
+      </names>
+    </group>
+  </macro>
+  <macro name="accessed">
+    <group delimiter=" ">
+      <choose>
+        <if type="graphic personal_communication interview" match="any">
+          <text variable="URL" text-decoration="none" prefix=" URL : " suffix="."/>
```

Comment:
you'll want to use non-breaking spaces before colons. Use `&160;` instead of the space here and otherwise (else you can get the : at the beginning of a new line).